core/templates/domain/skill/concept-card-backend-api.service.ts

```diff
@@ -17,92 +17,101 @@
  * about the concept card of a skill from the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/skill/skill-domain.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
 
-angular.module('oppia').factory('ConceptCardBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService', 'CONCEPT_CARD_DATA_URL_TEMPLATE',
-  function($http, $q, UrlInterpolationService, CONCEPT_CARD_DATA_URL_TEMPLATE) {
-    // Maps previously loaded concept cards to their IDs.
-    var _conceptCardCache = [];
+import cloneDeep from 'lodash/cloneDeep';
 
-    var _fetchConceptCards = function(
-        skillIds, successCallback, errorCallback) {
-      var conceptCardDataUrl = UrlInterpolationService.interpolateUrl(
-        CONCEPT_CARD_DATA_URL_TEMPLATE, {
-          comma_separated_skill_ids: skillIds.join(',')
-        });
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { SkillDomainConstants } from
+  'domain/skill/skill-domain.constants';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConceptCardBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolation: UrlInterpolationService) {}
 
-      $http.get(conceptCardDataUrl).then(function(response) {
-        var conceptCards = angular.copy(response.data.concept_card_dicts);
+  // Maps previously loaded concept cards to their IDs.
+  private _conceptCardCache = [];
+
+  private _fetchConceptCards(
+      skillIds: Array<string>,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var conceptCardDataUrl = this.urlInterpolation.interpolateUrl(
+      SkillDomainConstants.CONCEPT_CARD_DATA_URL_TEMPLATE, {
+        comma_separated_skill_ids: skillIds.join(',')
+      });
+
+    this.http.get(conceptCardDataUrl).toPromise().then(
+      (response: any) => {
         if (successCallback) {
-          successCallback(conceptCards);
+          successCallback(response.concept_card_dicts);
```

Comment:
Also as decided, the backend api services migrated from now on should return a frontend object with `camelCase` keys. So, please create an Object Factory for concept cards and use it here.

/cc @bansalnitish (Can you please take care of it in the future Angular Migration PRs)

Reply:
Fixed, I updated the service & unit tests to return an array of ConceptCard objects instead of an array of dicts.